_posts/2015-10-29-code-talks-java.md

```diff
@@ -0,0 +1,72 @@
+
+Besides advertising *jWebSockets* there was a benchmark showing that *Nashorn* (the new JavaScript engine coming with Java8) is way faster than the old *Rhino* (Java6/7) implementation.
+While leveraging Java8's `invokedynamic` it takes some additional time to warm up but afterwards outruns rhino for good.
+With *warming up* the "compiler people" refer to having the programme optimised by the machine until it runs several 10 times faster then in a cold state.
+And honestly the difference was impressive.
+That being said I noticed that `V8`, Google's JavaScript engine powering *nodejs* as well as Chrome, still being **twice** as fast as the warmed up Nashorn.
+Now that's something.
+
+Anyway, one has to keep in mind that Google threw a lot of resources into developing this awesome piece of software.
+Of course you don't get all of the JVM's nifty little things like concurrency, sandboxing and... well basically the complete Java stack but there have to be some drawbacks, right?
+To overcome this single-process-problem, node recently introduced a completely overhauled `cluster` module.
+This native module as well as process managers like *pm2* and *strongloop* do a great job in distributing load between several single-threaded node processes (and with that still crushing any JVM based JS engine).
+
+Nonetheless, the talk was pretty interesting with regards to what is possible with the JVM and how easy it is to run JavaScript code on a "Java Server" (I was again reminded why Java is called a typed language: Because you have to type so freaking much!).
+Things to keep in mind though: All used `npm` packages must be "nashorn compatible".
+Whatever this means.
+
+For reading this far you deserve a well kept secret about `npm`: It's **not** an abbreviation for the **n**ode **p**ackage **m**anager but a recursive acronymic abbreviation for "npm is not an acronym".
+
+However rumor has it, the package manager sends "**n**aughty **p**ush **m**essages" once installed (see [npm-expansions](https://github.com/npm/npm-expansions)).
+
+# 2. We're pretty much up-to-date
+
+In the Frontend/JavaScript track there were several talks covering various aspects.
+From the language itself ('There is no JavScript' - awesome presentation by Noam Kfir) to pretty interesting testing approaches and the life of suffering with React and ES6.
+We realised that our current testing and build pipeline which consists of `mocha`, `babel`, `karma` and `webpack` is not only considered state of the art but is just discovered by some companies.
+They shared knowledge of pitfalls they encountered and how they overcame them (or at least what they tried to do so).
+Their solution was basically our current build configuration.
+But since the topic is pretty tricky, there were of course some nifty tricks we've yet to adopt so we're very glad they went through all that pain and shared these valuable findings.
+Anyway it still gave us a pretty confident feeling about how we're doing things.
```

Comment:
Anyway, it still gave us a pretty confident feeling about how we're doing things.